src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -96,6 +102,16 @@ public static bool UseLazyResolution(MethodDesc method, string importModule, PIn
             if (forceLazyResolution.HasValue)
                 return forceLazyResolution.Value;
 
+
+            PInvokeMetadata metadata = method.GetPInvokeMethodMetadata();
+            for (int i = 0; i < s_lazyPInvokeDependencies.Length; i++)
+            {
+                if (s_lazyPInvokeDependencies[i] == metadata.Module)
```

Comment:
If this actually ever evaluates to true, we might have a bug somewhere. Feels like this should be `metadata.Name`.